### DISCUSSION TITLE
Promotion rule on item total does not include possible discount

### DISCUSSION THREAD

**Discussion by loicginoux** [solidusio/solidus#4601] [New Features or Ideas]
### Steps to reproduce
- create a promotion A that is eligible after 50$
- create a promotion B on a 60$ product that gives - 50%.
- add the product to the cart
- the promotion A is eligible even if the total of the cart after discount is 30$.

### expected behaviour
We expected item total rule to be calculated on the final price price the user see on his cart. He does not seem logical from its point of view if we tell him "you get free shipping after 50$" and he see the promo applied even if its cart is at 30$.

We actually changed the `Spree::Promotion::Rules::ItemTotal#eligible?` method in our app and calculate eligibility after all promotions applied (on line item and order). 

If you think that this behaviour must be kept in a per app basis, you can close this issue.

### System configuration
**Solidus Version**:
2.4

**Comment by jacobherrington** [comment]
This is a kind of gray question. Really depends on the shop owner I think, however I would think that ideally eligibility should be calculated after all other promotions are applied.
